packages/netlify-cms-core/src/backend.ts

```diff
@@ -493,28 +525,65 @@ export class Backend {
   async getEntry(state: State, collection: Collection, slug: string) {
     const path = selectEntryPath(collection, slug) as string;
     const label = selectFileEntryLabel(collection, slug);
-
+    const extension = selectFolderEntryExtension(collection);
     const integration = selectIntegration(state.integrations, null, 'assetStore');
+    const multiContent = collection.get('multi_content');
+    const locales = state.config.get('locales');
+    let loadedEntries;
+    let mediaFiles;
+
+    if (locales && multiContent === SAME_FOLDER) {
+      loadedEntries = await Promise.all(
+        locales.map(l =>
+          this.implementation
+            .getEntry(path.replace(extension, `${l}.${extension}`))
+            .catch(() => undefined),
+        ),
+      );
+    } else if (locales && multiContent === DIFF_FOLDER) {
+      loadedEntries = await Promise.all(
+        locales.map(l =>
+          this.implementation
+            .getEntry(path.replace(`${slug}`, `${l}/${slug}`))
+            .catch(() => undefined),
+        ),
+      );
+    } else {
+      const loadedEntry = await this.implementation.getEntry(path);
+      loadedEntries = [loadedEntry];
+    }
 
-    const loadedEntry = await this.implementation.getEntry(path);
-    const entry = createEntry(collection.get('name'), slug, loadedEntry.file.path, {
-      raw: loadedEntry.data,
-      label,
-      mediaFiles: [],
-    });
+    const entries = await Promise.all(
+      loadedEntries.filter(Boolean).map(async loadedEntry => {
+        const entry = createEntry(collection.get('name'), slug, loadedEntry.file.path, {
+          raw: loadedEntry.data,
+          label,
+          mediaFiles: [],
+        });
 
-    const entryWithFormat = this.entryWithFormat(collection)(entry);
-    const mediaFolders = selectMediaFolders(state, collection, fromJS(entryWithFormat));
-    if (mediaFolders.length > 0 && !integration) {
-      entry.mediaFiles = [];
-      for (const folder of mediaFolders) {
-        entry.mediaFiles = [...entry.mediaFiles, ...(await this.implementation.getMedia(folder))];
-      }
-    } else {
-      entry.mediaFiles = state.mediaLibrary.get('files') || [];
+        const entryWithFormat = this.entryWithFormat(collection)(entry);
+        if (!mediaFiles) {
+          const mediaFolders = selectMediaFolders(state, collection, fromJS(entryWithFormat));
+          if (mediaFolders.length > 0 && !integration) {
+            mediaFiles = [];
+            for (const folder of mediaFolders) {
+              mediaFiles = [...entry.mediaFiles, ...(await this.implementation.getMedia(folder))];
```

Comment:
I think we can read the media files just for the default locale and not for each locale right?

Reply:
With this check https://github.com/netlify/netlify-cms/pull/3366/files/7a5df4f3dac15ad5beec5e67610dd67870f3494e#diff-98047a59f166f83c042f0f4d7589d550R565, it will be read once.

Reply:
Oh, missed that